pkg/controllers/autoimport/autoimport_controller.go

```diff
@@ -0,0 +1,306 @@
+			return ctrl.Result{}, nil
+		}
+		if err := ensureAutoImportSecret(r, managedClusterName, &kubeconfig); err != nil {
+			return ctrl.Result{}, err
+		}
+	}
+
+	return ctrl.Result{}, ensureCreateManagedClusterAnnotationFalse(r, &hyd)
+}
+
+func (r *Reconciler) SetupWithManager(mgr ctrl.Manager) error {
+	syncedFromSpoke := func(object client.Object) bool {
+		return strings.EqualFold(object.GetLabels()["synced-from-spoke"], "true")
+	}
+
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&hypdeployment.HypershiftDeployment{}).
+		// TODO(zhujian7): After https://github.com/stolostron/hypershift-deployment-controller/pull/33 is merged,
+		// we can add a new secret controller to render the kubeConfig in status and remove the watches here.
+		Watches(&source.Kind{Type: &corev1.Secret{}}, handler.EnqueueRequestsFromMapFunc(func(obj client.Object) []reconcile.Request {
+			an := obj.GetAnnotations()
+			if len(an) == 0 || len(an[constant.AnnoHypershiftDeployment]) == 0 {
+				return []reconcile.Request{}
+			}
+
+			res := strings.Split(an[constant.AnnoHypershiftDeployment], constant.NamespaceNameSeperator)
+			if len(res) != 2 {
+				log.Log.Error(fmt.Errorf("failed to get hypershiftDeployment"), "annotation invalid",
+					"constant.AnnoHypershiftDeployment", an[constant.AnnoHypershiftDeployment])
+				return []reconcile.Request{}
+			}
+
+			req := reconcile.Request{
+				NamespacedName: types.NamespacedName{Namespace: res[0], Name: res[1]},
+			}
+			return []reconcile.Request{req}
+		}), builder.WithPredicates(
+			predicate.Funcs{
+				GenericFunc: func(e event.GenericEvent) bool { return false },
+				CreateFunc: func(e event.CreateEvent) bool {
+					return syncedFromSpoke(e.Object)
+				},
+				DeleteFunc: func(e event.DeleteEvent) bool { return false },
+				UpdateFunc: func(e event.UpdateEvent) bool {
+					if !syncedFromSpoke(e.ObjectNew) {
+						return false
+					}
+
+					new, okNew := e.ObjectNew.(*corev1.Secret)
+					old, okOld := e.ObjectOld.(*corev1.Secret)
+					if okNew && okOld {
+						return !equality.Semantic.DeepEqual(old.Data, new.Data)
+					}
+					return false
+				},
+			},
+		)).
+		WithOptions(controller.Options{
+			MaxConcurrentReconciles: 1, // This is the default
+		}).Named("hypershiftimport").Complete(r)
+}
+
+func ensureManagedCluster(r *Reconciler, hydNamespaceName types.NamespacedName,
+	managedClusterName, managementClusterName string) (*mcv1.ManagedCluster, error) {
+	log := r.Log.WithValues("managedClusterName", managedClusterName)
+	ctx := context.Background()
+
+	var mc mcv1.ManagedCluster
+	err := r.Get(ctx, types.NamespacedName{Name: managedClusterName}, &mc)
+	if k8serrors.IsNotFound(err) {
+		log.V(INFO).Info("Create a new ManagedCluster resource")
+		mc.Name = managedClusterName
+		mc.Spec.HubAcceptsClient = true
+
+		mc.ObjectMeta.Annotations = map[string]string{
+			"import.open-cluster-management.io/klusterlet-deploy-mode":  "Detached",
```

Comment:
I thought we were changing this name?

Reply:
yeah, After the upstream ocm changes(ASAP) this to `Hosted`, I will update it here.